pkg/ipamd/ipamd.go

```diff
@@ -356,6 +368,12 @@ func (c *IPAMContext) nodeInit() error {
 		retry := 0
 		for {
 			retry++
+
+			if c.awsClient.IsCNIUnmanagedENI(eni.ENIID) {
+				log.Infof("Skipping ENI %s since it is not on network card 0", eni.ENIID)
+				continue
+			}
+
```

Comment:
Have we tested this workflow?

Does not look like the right place to have this check and the `continue` statement. Seems like for a given ENI that is not on network card 0, this loop will be keep spinning.

You would want to move this outside the retry loop

Reply:
Yeah you are right, it should have been a break, because it will run until -

```
if retry > maxRetryCheckENI {
				log.Warnf("Reached max retry: Unable to discover attached IPs for ENI from metadata service (attempted %d/%d): %v", retry, maxRetryCheckENI, err)
				ipamdErrInc("waitENIAttachedMaxRetryExceeded")
				break
			}
```

Reply:
Nvm, this needs to be fixed. Thank you. Good catch.